### PR TITLE
Product views

### PR DESCRIPTION
**Pull request by zstephenj**
Mini-testing guide for this PR:
http://localhost:8080/dashboard
http://localhost:8080/product/1
http://localhost:8080/product/2
http://localhost:8080/product/3
http://localhost:8080/productsearchresult

- Our components are starting to stack up in number, so I created a Products folder inside of Components and refactored import statements.

- For ease of testing, I populate state.Products.AllProducts from a fakeJSON data server in the created() hook. The SingleProduct pages draw on a database with only three products (ids 1-3). ProductSearchResult and the FarmerDashboard draws on a different database with many more products. Just know that some links may not work, but are directed at the right places.

- Adds gradient and navy font-color styling to App.vue

- Adds editProduct action and mutation to Vuex Store

- Adds a FarmerProductDash component to FarmerDashboard that displays a table of products

- FarmerProductDash can quickly edit Price/Quantity, routes to SingleProduct/EditProduct pages, removes items

- Adds SingleProduct view 

- Adds ProductSearchResult

### CODE REVIEW

File: frontend/src/components/Product/AddProduct.vue [hunk 1]
```diff
@@ -251,12 +262,17 @@ export default {
             return true
         },
         
-        generateBase64(img) {
+        
+        generateBase64() {
```
Comment (ryanis-online) [inline frontend/src/components/Product/AddProduct.vue]: Very interesting solution to encode and images!

**Review by ryanis-online — APPROVED** [review]
Nice use of the fake JSON server to test these. Good work 👍

**Review by iktani — APPROVED** [review]
I like the quick edit functionality in the farmer dashboard. Very cool! Single product and product search views look good. Nicely Done! Deleting from the dashboard and editing when I tested weren't working but I think you're still tinkering with that.

**Comment by zstephenj** [comment]
Just pushed out fixes to everything. All pages should display properly. I broke the SingleProduct view and EditProduct for a few hours today. @iktani  Nice call on the delete. I forgot to handle that functionality.

**Comment by zstephenj** [comment]
I know this was outside the scope of this PR but you might also check out the new AddProduct and EditProduct pages. EditProduct will fill in all values in the form, given the id parameter.

**Pull request merged**